### DISCUSSION TITLE
Updates to steps taking some time to reflect in validation

### DISCUSSION THREAD

**Discussion by aedwards9** [getgauge/gauge-vscode#946] [General]
When the step implementation files are updated (e.g. via a git pull) it can sometimes take a little while for the Gauge VSCode plugin to pick up the new steps and thus can sometimes show errors in a specification (saying the step is invalid) when it isn't. This normally resolves itself eventually but I wondered if there is a way to "force" this to happen. I think normally closing/opening code works but would be nice to have something quicker. I'm not sure how the validation is happening behind the scenes but is there a way I can force it to reset / re-validate within the VSCode?

**Comment by jensakejohansson** [comment]
I've noticed this issue as well. As you say, I do reload the project when this happens:
<img width="1069" height="63" alt="image" src="https://github.com/user-attachments/assets/c65ab4e8-9983-44f9-862e-76410398aa08" />

A not so nice (and in large project slow) work around. I think your idea is good, implemented a command the quickly forces a 'fresh reload' of the validation when this happens. Or solve the root cause of the issue. I might investigate when/if i find the time...

**Comment by jensakejohansson** [comment]
@aedwards9 What language are you using for implementing your steps? C#?
